### DISCUSSION TITLE
session_request, session_update hooks for appkit [feature]

### DISCUSSION THREAD

**Discussion by VGeorgiev1** [reown-com/appkit#3705] [Ideas]
### What problem does this new feature solve?

I have been creating a dApp with reown appKit and I really need to be able to execute some logic when a signing request for a transaction is initiated. I found that the WalletKit provides a hook for listening for the session_request event but I cant find a way of using it with appKit.
Its really confusing because when I reject a signing request i get the following error:
Object Error: emitting session_request:1733920443898539 without any listeners which makes me think that appkit already listens for those events but there are no exposed hooks for them. 

### Describe the solution you'd like

Easy way to register an event listenr for `session_update` `session_request` events etc.
Example: `appKit.on('session_update', () => {})`

[a comment by an automated account is not shown]

**Comment by rtomas** [comment]
Thanks for your feature request. We will be analysing with the team.
Would you mind sharing what you are building ?

**Comment by VGeorgiev1** [comment]
I would be able to share that, but I can assure you I have checked all of the web examples and there is no such feature or a close one to that in the AppKit.

**Comment by arein** [comment]
We need to have a crisp explanation of your use case please do be able to cross-check whether we don't support that with the existing APIs and also in case we don't to be able to properly prioritize

> I really need to be able to execute some logic when a signing request for a transaction is initiated

@VGeorgiev1 your app is initiating the signing request - aka usually user clicks a button in your app and then you trigger the signing request. So it's unclear why you need to hook into things lower level - please explain

**Comment by narutodev123** [comment]
I also got stucked with this issue i want to instantiate all my api call related to wallet address once the session is created successfully is there any hook or listener to identify when SWIX session is created.
